smartapps/opent2t/opent2t-smartapp-test.src/opent2t-smartapp-test.groovy

```diff
@@ -0,0 +1,355 @@
+        ]
+    }
+    path("/connectionId/:connId") {
+        action: [
+            PUT: "updateConnectionId"
+        ]
+    }
+	path("/devices") {
+		action: [
+			GET: "getDevices"
+		]
+	}
+    path("/devices/:id") {
+		action: [
+			GET: "getDevice"
+		]
+	}
+    path("/update/:id") {
+        action: [
+          PUT: "updateDevice"
+        ]
+    }
+}
+
+def installed() {
+	log.debug "Installed with settings: ${settings}"
+	initialize()
+}
+
+def updated() {
+	log.debug "Updated with settings: ${settings}"
+	unsubscribe()
+	initialize()
+}
+
+def initialize() {
+	state.connectionId = ""
+    state.endpointURL = "https://ifs.windows-int.com/v1/cb/81C7E77B-EABC-488A-B2BF-FEC42F0DABD2/notify"
+	registerSubscriptions()
+}
+
+//Register events for each device
+def registerSubscriptions() {
+	registerChangeHandler(switches)
+	registerChangeHandler(motionSensors)
+    registerChangeHandler(presenceSensors)
+    registerChangeHandler(contactSensors)
+    registerChangeHandler(waterSensors)
+    registerChangeHandler(locks)
+    registerChangeHandler(garageDoors)
+    registerChangeHandler(cameras)
+}
+
+def registerChangeHandler(myList) {
+	myList.each { myDevice ->
+		def theAtts = myDevice.supportedAttributes
+		theAtts.each {att ->
+		    subscribe(myDevice, att.name, eventHandler)
+    	log.debug "Registering ${myDevice.displayName}.${att.name}"
+		}
+	}
+}
+
+//When events are triggered, send HTTP post to web socket servers
+def eventHandler(evt) {
+
+	def evt_device_id = evt.deviceId
+    def evt_device_value = evt.value
+    def evt_name = evt.name
+    def evt_device = evt.device
+    def evt_deviceType = getDeviceType(evt_device);
+    def params = [
+    	uri: "${state.endpointURL}/${state.connectionId}",
+        body: [
+            name: evt_device.displayName,
+            id: evt_device.id,
+            deviceType:evt_deviceType, 
+            attributes: deviceAttributeList(evt_device) 
+        ]
+    ]
+    try {
+        log.debug "POST URI: ${params.uri}"
+        log.debug "Payload: ${params.body}"
+        httpPostJson(params) { resp ->
+            resp.headers.each {
+                log.debug "${it.name} : ${it.value}"
+            }
+            log.debug "response status code: ${resp.status}"
+            log.debug "response data: ${resp.data}"
+        }
+    } catch (e) {
+        log.debug "something went wrong: $e"
+    }
+}
+
+//Endpoints function; update subcription endpoint url [state.endpoint]
+void updateEndpointURL() {
+    state.endpointURL = params.url
+    log.debug "Updated EndpointURL to ${state.endpointURL}"
+}
+
+//Endpoints function; update global variable [state.connectionId]
+void updateConnectionId() {
+    def connId = params.connId
+    state.connectionId = connId
+    log.debug "Updated ConnectionID to ${state.connectionId}"
+}
+
+//Endpoints functions; return data from all devices supported in JSON format
+def getDevices() {
+	def deviceData = [] 
+    switches.each {
```

Comment:
If below are all the same, you can replace them all with one call (IF you will only use input elements with devices!):

inputs?.each {
     	def deviceType = getDeviceType(it)
         deviceData << [name: it.displayName, id: it.id, deviceType:deviceType, attributes: deviceAttributeList(it)] 
     }